src/pybind/mgr/diskprediction_local/predictor.py

```diff
@@ -39,227 +38,158 @@ class DiskFailurePredictor(object):
+                                                features from one day.
 
         Returns:
-            modelpath: The best model for the given attribute list.
-            model_attrlist: 'Ordered' attribute list of the returned model.
-                            Must be aware that SMART attributes is in order.
-
-        Raises:
+            numpy.ndarray -- (n, d) shaped array of n days worth of data and d
+                                features, scaled
         """
-
-        models = self.model_context.keys()
-
-        scores = []
-        for model_name in models:
-            scores.append(sum(attr in attr_list
-                              for attr in self.model_context[model_name]))
-        max_score = max(scores)
-
-        # Skip if too few matched attributes.
-        if max_score < 3:
-            print("Too few matched attributes")
+        # preprocessing may vary across manufactueres. so get manufacturer
+        manufacturer = DiskFailurePredictor.__get_manufacturer(disk_days_df['model_name'].iloc[0]).lower()
+
+        # keep only the features used for prediction for current manufacturer
+        try:
+            disk_days_df = disk_days_df[self.model_context[manufacturer]]
+        except KeyError as e:
+            # TODO: change to log.error
+            print("Either SMART attributes mismatch for hard drive and prediction model,\
+                 or 'model_name' not available in input data")
+            print(e)
             return None
 
-        best_models = {}
-        best_model_indices = [idx for idx, score in enumerate(scores)
-                              if score > max_score - 2]
-        for model_idx in best_model_indices:
-            model_name = list(models)[model_idx]
-            model_path = os.path.join(self.model_dirpath, model_name)
-            model_attrlist = self.model_context[model_name]
-            best_models[model_path] = model_attrlist
+        # scale raw data
+        preprocessor_path = os.path.join(self.model_dirpath, manufacturer + '_preprocessor.joblib')
+        preprocessor = joblib.load(preprocessor_path)
+        disk_days_df = preprocessor.transform(disk_days_df)
+        return disk_days_df
 
-        return best_models
-        # return os.path.join(self.model_dirpath, model_name), model_attrlist
 
     @staticmethod
-    def __get_ordered_attrs(disk_days, model_attrlist):
-        """
-        Return ordered attributes of given disk days.
+    def __get_manufacturer(model_name):
+        """Returns the manufacturer name for a given hard drive model name
 
-        Args:
-            disk_days: Unordered disk days.
-            model_attrlist: Model's ordered attribute list.
+        Arguments:
+            model_name {str} -- hard drive model name
 
         Returns:
-            ordered_attrs: Ordered disk days.
-
-        Raises: None
+            str -- manufacturer name
         """
+        if model_name.startswith("W"):
```

Comment:
can we instead put:
```py
vendors = {'WDC': 'WDC',
           'TOSHIBA': 'Toshiba',
           'ST': 'Seagate',
           'Hitachi': 'Hitachi',
           'HGST': 'HGST'}
for prefix, vendor in vendors:
  if model_name.startwith(prefix):
    return vendor
raise LookupError('unknown manufacturer of model {}'.format(model_name))

```

Reply:
Absolutely. Although seagate seems to have more than one prefixes (at least in the backblaze dataset - see "Out[8]" in [this](https://github.com/chauhankaranraj/ceph_drive_failure/blob/master/dask_data_explorer.ipynb) jupyter notebook). Do you think this would work;
```python3
vendors = {'WDC': 'WDC',
           'Toshiba': 'Toshiba',
           'S': 'Seagate',     # for cases like "STxxxx" and "Seagate BarraCuda ZAxxx"
           'ZA': 'Seagate',    # for cases like "ZAxxxx"
           'Hitachi': 'Hitachi',
           'HGST': 'HGST'}
for prefix, vendor in vendors.items():
  if model_name.startswith(prefix):
    return vendor
raise LookupError('unknown manufacturer of model {}'.format(model_name))
```
Would you happen to know if this is the case with other vendors as well (i.e. same vendor but different model name prefixes)?

Reply:
@tchaikov when the manufacturer is unknown, should it error out and stop execution (`raise LookupError`) or should it return "Unknown"

Reply:
> Would you happen to know if this is the case with other vendors as well

@chauhankaranraj no, i was just using the source code of SMART as a reference.

> when the manufacturer is unknown, should it error out and stop execution 


good question indeed. probably we can use a fallback. i.e., to print out an error message containing the model name then continue on?

Reply:
Gotcha, thanks. I updated the code as per your suggestion.